tests/v2/validation/provisioning/k3s/post_kdm_oob_release_test.go

```diff
@@ -0,0 +1,159 @@
+package k3s
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	kubeProvisioning "github.com/rancher/rancher/tests/framework/clients/provisioning"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	provisioningV1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters/versions"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type KdmChecksTestSuite struct {
+	suite.Suite
+	session               *session.Session
+	client                *rancher.Client
+	ns                    string
+	k3skubernetesVersions []string
+	cnis                  []string
+	providers             []string
+	nodesAndRoles         []machinepools.NodeRoles
+}
+
+const (
+	defaultNamespace = "default"
+)
+
+func (k *KdmChecksTestSuite) TearDownSuite() {
+	k.session.Cleanup()
+}
+
+func (k *KdmChecksTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	k.session = testSession
+
+	k.ns = defaultNamespace
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	k.k3skubernetesVersions = clustersConfig.K3SKubernetesVersions
+
+	k.cnis = clustersConfig.CNIs
+	k.providers = clustersConfig.Providers
+	k.nodesAndRoles = clustersConfig.NodesAndRoles
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(k.T(), err)
+
+	k.client = client
+}
+
+func (k *KdmChecksTestSuite) TestK3SK8sVersions() {
+	logrus.Infof("checking for valid k8s versions..")
+	require.GreaterOrEqual(k.T(), len(k.k3skubernetesVersions), 1)
+	// fetching all available k8s versions from rancher
+	releasedK8sVersions, _ := versions.ListK3SAllVersions(k.client)
```

Comment:
I have a question. 

Do we expect that the list of all k8s/k3s to never miss existing versions? 

I am thinking these tests here only validate expected OOB and assumes the others supported versions are in the list.

Reply:
ListK3SAllVersion list all the k8s version that's present in the drop down of rancher server.
And yes, whatever k8s version we need to verify we can pass that in the cattle-config.yaml.